sktime/forecasting/model_selection/_split.py

```diff
@@ -186,12 +190,24 @@ def _get_end(y: ACCEPTED_Y_TYPES, fh: ForecastingHorizon) -> int:
     # For purely in-sample forecasting horizons, the last split point is the end of the
     # training data.
     if fh.is_all_in_sample():
-        end = n_timepoints + 1
+        if array_is_int(fh):
+            end = n_timepoints - 1
```

Comment:
wait, why is it "minus 1" here if it is "plus one" in the old version??

Reply:
I have been struggling with this part a lot. The "old version" was totally confusing to me. So I have changed to to what you see now with the following comment in the docstring:
```
For a time series `y`, `y.iloc[end]` will give
the last value of the training window,
while `y.index[end]` will provide the last index of the training window.
```
Let me know if you disagree, that this interpretation/intention of `_get_end` is much more understandable.

In addition, of course, I have compensated this change in `_get_end` output by adding 2 accordingly downstream.

Reply:
why "plus two", though, can you please explain that? Still don't understand.

Reply:
The original formula was `end = n_timepoints + 1`. Now it is `end = n_timepoints - 1`. The difference is 2. Let's reverse the question: do you understand the original formula? I do not.